### ISSUE TITLE
Skills that go through walls but shouldn't

### ISSUE THREAD

**Issue by AnisotropicDefixation** [HerculesWS/Hercules#1175]
RK_WINDCUTTER
RK_IGNITIONBREAK
RK_DRAGONBREATH
RK_DRAGONBREATH_WATER
NC_COLDSLOWER

**Comment by AnisotropicDefixation** [comment]
Feel free to add missing skills.

**Comment by Playtester** [comment]
Are they even any player-casted damage skills that actually go through walls at all on official servers?

**Comment by AnisotropicDefixation** [comment]
I'm not sure, but about everything does on emulators.

**Comment by Playtester** [comment]
Yeah it's one of the biggest problems on emulator imo, it's also really bad to add these checks for every single skill individually rather than just implement the common rule.

I investigates this for rAthena too, you can read what I found here:
https://github.com/rathena/rathena/issues/937#issue-129856301
(I summarized everything in the first post.)

**Comment by AnisotropicDefixation** [comment]
Sadly, foreachinshootrange doesn't work properly for certain skills.

**Comment by Rytech2** [comment]
Bumping this up so ill remember to check into this,

**Comment by Playtester** [comment]
Related rAthena commit: https://github.com/rathena/rathena/commit/7906f0a3ff3742bb519b19b0cadbca9db0d35b32

Note: I only checked and fixed the skills mentioned in the commit message. I didn't have the time to test and fix all the 3rd class skills.

On a sidenote, Dragon Breath and Arrow Shower are different from all the other skills in that they can actually hit through walls if it's a one-tile wall and you target the wall directly. This is proof that they can't be unit skills as units couldn't be placed on walls. Compare to Thunder Storm which doesn't hit at all when you cast it on a wall.
